docs/e2e-test.md

```diff
@@ -102,3 +102,14 @@ For instance, to skip the upgrade E2E tests set `GINKGO_SKIP="[upgrade]"`
 Both Ironic and BMO upgrade tests currently only check that a new version can be rolled out (by going from `latest` to `main`).
 However, they do not actually upgrade from some older release to a newer since they are not yet integrated with the e2e upgrade test.
 The idea is to move them to the e2e upgrade test and then upgrade Ironic and BMO together with CAPM3 from the previous minor release to the latest.
+
+### Test matrix for k8s version
+
+Current e2e tests use the following Kubernetes versions for source and target clusters:
```

Comment:
I know we have this terminology everywhere, but I think we should try to use "bootstrap", "management", "workload" and "self-hosted" terminology instead of "source" and "target". In CAPI "source" and "target" is only used in the context of pivoting. In other contexts it is quite confusing to use these terms.

We start with the bootstrap cluster, we pivot to the target management cluster that is self-hosted.

```suggestion
Current e2e tests use the following Kubernetes versions for bootstrap and self-hosted management clusters:
```

Reply:
Ah even more reason to not say target cluster! What about "CAPI cluster"? Or we can say "workload/management" cluster

Reply:
as management can also refer too bootstrap I suggest use metal3 cluster or metal3 provisioned cluster

Reply:
this is overcomplicating @mboukhalfa , anyways lets take this PR in so that I can test e2e with correct k8s versions and lets debate on the name of the cluster in a followup I would do soon.